python/lsst/ctrl/bps/wms/panda/idds_tasks.py

```diff
@@ -0,0 +1,126 @@
+class LSSTTask(object):
+    name = None
+    step = None
+    queue = None
+    executable = None
+    maxwalltime = None
+    maxattempt = None
+    lfns = []
+    local_pfns = []
+    dependencies = []
+
+
+class IDDSWorkflowGenerator:
+    def __init__(self, bps_workflow, config):
+        self.bps_workflow = bps_workflow
+        self.bps_config = config
+        self.tasks_inputs = {}
+        self.himem_tasks = set(config.get("himem_steps"))
+        self.computing_queue = config.get("computing_queue")
+        self.computing_queue_himem = config.get("computing_queue_himem")
+
+        _, v = config.search("maxwalltime", opt={"default": 90000})
+        self.maxwalltime = v
+        _, v = config.search("maxattempt", opt={"default": 5})
+        self.maxattempt = v
+
+    def define_task_name(self, step):
+        return self.bps_config.get('workflowName') + '_' + step
+
+    def pick_non_init_cmdline(self):
+        for node_name in self.bps_workflow.nodes:
+            if node_name != 'pipetaskInit':
+                return self.bps_workflow.nodes[node_name]['job'].cmdline
+
+    def define_tasks(self):
+        tasks = []
+        raw_dependency_map = self.create_raw_jobs_dependency_map()
+        tasks_dependency_map = self.split_map_over_tasks(raw_dependency_map)
+        init_task_cmd_line = self.bps_workflow.nodes['pipetaskInit']['job'].cmdline
+        other_task_cmd_line = self.pick_non_init_cmdline()
+
+        for task_step, jobs in tasks_dependency_map.items():
+            task = LSSTTask()
+            task.step = task_step
+            task.name = self.define_task_name(task.step)
+            task.queue = self.computing_queue_himem if task_step in self.himem_tasks else self.computing_queue
+            task.lfns = list(jobs.keys())

```

Comment:
if `jobs` is dict-like then `list(jobs)` is shorter and results in the same thing.

Reply:
Thanks. Fixed.